browser/src/control/Permission.js

```diff
@@ -84,18 +90,27 @@ L.Map.include({
 	},
 
 	_getFileExtension: function (filename) {
-		return filename.substring(filename.lastIndexOf('.'));
+		return filename.substring(filename.lastIndexOf('.') + 1);
 	},
 
-	_isFilePlainText: function () {
+	_shouldStartReadOnly: function () {
 		var fileName = this['wopi'].BaseFileName;
 		// use this feature for only integration.
 		if (!fileName) return false;
 		var extension = this._getFileExtension(fileName);
-		return extension === '.csv' || extension === '.txt';
+		if (!Object.prototype.hasOwnProperty.call(this.readonlyStartingFormats, extension))
+			return false;
+		return true;
 	},
 
 	_proceedEditMode: function() {
+		var fileName = this['wopi'].BaseFileName;
+		if (fileName) {
+			var extension = this._getFileExtension(fileName);
+			var extensionInfo = this.readonlyStartingFormats[extension];
+			if (!extensionInfo.canEdit)
```

Comment:
extensionInfo can be `undefined` I think for formats not present in `readonlyStartingFormats`

Reply:
ok, false alarm, this is called only when `_shouldStartReadOnly` is true...